record_test.go

```diff
@@ -156,6 +156,29 @@ func TestParseRecord(t *testing.T) {
 			},
 			status: 404,
 		},
+		{
```

Comment:
http headers are case insensitive. We should test this insensitivity and make it so.

Reply:
Ref: http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2

Reply:
Aww completely missed that. I guess we have to loop through all the headers and lowercase their key and check them with the header provided in the record.

Reply:
+1